packages/components/src/components/input/input.tsx

```diff
@@ -1,11 +1,49 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import Field from '../field';
 import Text from '../text/text';
 
-const InputWrapper = ({ children, has_footer }) =>
-    has_footer ? <div className='dc-input__wrapper'>{children}</div> : children;
+type TInputProps = {
+    autoComplete?: string;
+    bottom_label?: string;
+    className?: string;
+    classNameError?: string;
+    classNameHint?: string;
+    classNameWarn?: string;
+    data_testId?: string;
+    disabled?: boolean;
+    error?: string;
+    field_className?: string;
+    has_character_counter?: boolean;
+    hint?: React.ReactNode;
+    id?: string;
+    initial_character_count?: number;
+    input_id?: string;
+    is_relative_hint?: boolean;
+    label_className?: string;
+    label?: React.ReactNode;
+    leading_icon?: React.ReactElement;
+    max_characters?: number;
+    maxLength?: number;
+    name?: string;
+    onBlur?: (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+    onChange?: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+    onFocus?: (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+    onPaste?: (e: React.ClipboardEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+    placeholder?: string;
+    required?: boolean;
+    trailing_icon?: React.ReactElement;
+    type: string;
+    value?: string;
+    warn?: string;
+};
+
+type TInputWrapper = {
+    has_footer: boolean;
+};
+
+const InputWrapper = ({ children, has_footer }: React.PropsWithChildren<TInputWrapper>) =>
+    has_footer ? <div className='dc-input__wrapper'>{children}</div> : <React.Fragment>{children}</React.Fragment>;
 
 const Input = React.forwardRef(
```

Comment:
@jim-deriv `forwardRef` does accept generic types, Please pass the types in generic instead of manually typing the props 🙏🏻

Reply:
```suggestion
const Input = React.forwardRef<HTMLInputElement & HTMLTextAreaElement, TInputProps>(
```
@jim-deriv I guess @farzin-deriv is talking about this. Also, in this case you won't need to add types at lines 72,73 🙂